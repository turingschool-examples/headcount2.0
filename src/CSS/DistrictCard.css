.year-red {
  color: red;
}

.year-black {
  color: green;
}

.district-card {
  text-align: center;
  box-shadow: 2px 2px 2px 2px #ccc;
}

.district-location {
  font-size: 1.5rem;
}
