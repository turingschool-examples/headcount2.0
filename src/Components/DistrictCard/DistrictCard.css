.highStats {
  color: #7097B2;
}

.lowStats {
  color: #B81F38;
}

.card {
  border-radius: 18px;
  border: solid #4F4C47 2px;
  color: #4F4C47;
  display: inline-block;
  margin: 1rem;
  padding: .5rem;
  width: 300px;
}

.selected {
  background-color: #F7F5F1;
  border: solid #21201E 4px;
}

.card:hover {
  border: solid #4F4C47 4px;
}

.district {
  font-family: avenir, sans-serif;
  font-size: 1.5em;
  text-align: center;
}

.stats {
  font-family: avenir, sans-serif;
  font-size: 1.25em;
  padding-left: 5rem;
}