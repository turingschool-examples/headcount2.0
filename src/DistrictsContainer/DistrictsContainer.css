.DistrictsContainer {
	background-color: #E9EBEE;
	display: grid;
	grid-gap: 0.75rem;
	grid-template-columns: repeat(5, 1fr);
	margin: 0 5rem;
}

