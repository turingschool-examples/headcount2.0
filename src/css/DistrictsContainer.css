.district-container {
  display: grid;
  grid-template-columns: repeat(5, 1fr);
  justify-items: center !important;
}
