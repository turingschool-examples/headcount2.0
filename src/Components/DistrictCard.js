import React from "react";
import "../CSS/DistrictCard.css";
import PropTypes from "prop-types";

export const DistrictCard = ({
  location,
  stats,
  clicked,
  selectCard,
  counter
}) => {
  return (
    <div
      onClick={() => selectCard(location)}
      className={clicked ? "district-card-clicked" + counter : "district-card"}
    >
      <h1 className="district-location">{location}</h1>
      <header className={clicked ? "year-header-show" : "year-header-hide"}>
        Year
      </header>
      <header className={clicked ? "avg-header-show" : "avg-header-hide"}>
        Avg
      </header>
      {Object.keys(stats).map((year, index) => {
        const avg = stats[year];
        return (
          <section
            className={clicked ? "stats-show" : "stats-hide"}
            key={index}
          >
            <p className="year">
              {year}
              ......
              <span className={avg < 0.5 ? "avg-low" : "avg-high"}>{avg}</span>
            </p>
          </section>
        );
      })}
      <p className={clicked ? "show-msg" : "hide-msg"}>Click to Remove</p>
    </div>
  );
};

const { string, shape, number, bool, func } = PropTypes;

DistrictCard.propTypes = {
  location: string,
  stats: shape({ year: number }),
  clicked: bool,
  selectCard: func,
  counter: number
};
