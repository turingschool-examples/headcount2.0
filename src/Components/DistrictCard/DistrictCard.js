import React from 'react';
import './DistrictCard.css';
import PropTypes from 'prop-types';


const DistrictCard = ({district , selectDistrict}) => {
  const years = Object.keys(district.stats)

  const determineColor = (stats) => {
    return stats > 0.5 ? "highStats" : "lowStats"
  }

  const createStats = years.map((year, index) => {
    const stats = district.stats[year]
    return (
      <li className={determineColor(stats)} key={index}>
        {[year]}: {stats}
      </li>
      )
  })

  const handleClick = (location) => {
    selectDistrict(location)
  }

  return (
    <section 
      className={`card ${district.selected ? 'selected' : ''}`} 
      id={district.key}
      key={district.key}
      onClick={() => {handleClick(district.location)}}
    > 
      <h2 className="district">District Name: {district.location}</h2>
      <ul className="stats">
        {createStats}
      </ul>
    </section>
  )
}

DistrictCard.propTypes = { 
  district: PropTypes.object
}



export default DistrictCard;