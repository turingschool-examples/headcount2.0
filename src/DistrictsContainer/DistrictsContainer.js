import React from 'react';
import './DistrictsContainer.css';
import DistrictCard from '../DistrictCard/DistrictCard.js';
import PropTypes from 'prop-types';

const DistrictsContainer = ({ districts, compareDistrict, districtsBeingCompared }) => {
		const districtCards = districts.map((district, i) => {
		return <DistrictCard  
							key={i} 
							{...district}
							district={district}
							compareDistrict={compareDistrict}
							districtsBeingCompared={districtsBeingCompared}
						/>
	})

	return (
		<div className='DistrictsContainer'>
			{ districtCards }
		</div>
	)
}

DistrictsContainer.propTypes = {
	districts: PropTypes.array.isRequired,
	compareDistrict: PropTypes.func
}

export default DistrictsContainer;