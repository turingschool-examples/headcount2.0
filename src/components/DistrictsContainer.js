import React from "react";
import DistrictCard from "./DistrictCard";
import "../css/DistrictsContainer.css";
import PropTypes from "prop-types";

const DistrictsContainer = ({ districts, selectCards }) => {
  const districtCard = districts.map((district, index) => (
    <DistrictCard {...district} key={index} selectCards={selectCards} />
  ));
  return <div className="district-container">{districtCard}</div>;
};

DistrictsContainer.propTypes = {
  districts: PropTypes.array,
  selectCards: PropTypes.func
};

export default DistrictsContainer;
