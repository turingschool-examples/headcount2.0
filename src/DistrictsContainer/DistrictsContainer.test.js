import React from 'react';
import ReactDOM from 'react-dom';
import DistrictsContainer from './DistrictsContainer.js';
import DistrictCard from '../DistrictCard/DistrictCard.js';
import { shallow } from 'enzyme';

// it('renders without crashing', () => {
// 	const div = document.createElement('div');
// 	ReactDOM.render(<DistrictsContainer />, div)
// })

describe('DistrictsContainer', () => {
	let wrapper;
	let mockDistricts;

	beforeEach(() => {
			mockDistricts = [{'colorado': {}}, {'academy 20': {}}, {'agate 300': {}}] 
		wrapper = shallow(<DistrictsContainer districts={mockDistricts} key={Math.random()}/>)
	})

	it('matches the snapshot', () => {
		expect(wrapper).toMatchSnapshot()
	})

	it('renders each district', () => {
		expect(wrapper.find(DistrictCard).length).toEqual(3)
	})
})
