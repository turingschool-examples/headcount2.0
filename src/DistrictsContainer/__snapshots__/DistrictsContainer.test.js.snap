// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DistrictsContainer matches the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <DistrictsContainer
    districts={
      Array [
        Object {
          "colorado": Object {},
        },
        Object {
          "academy 20": Object {},
        },
        Object {
          "agate 300": Object {},
        },
      ]
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <DistrictCard
          colorado={Object {}}
          district={
            Object {
              "colorado": Object {},
            }
          }
        />,
        <DistrictCard
          academy 20={Object {}}
          district={
            Object {
              "academy 20": Object {},
            }
          }
        />,
        <DistrictCard
          agate 300={Object {}}
          district={
            Object {
              "agate 300": Object {},
            }
          }
        />,
      ],
      "className": "DistrictsContainer",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "0",
        "nodeType": "class",
        "props": Object {
          "colorado": Object {},
          "compareDistrict": undefined,
          "district": Object {
            "colorado": Object {},
          },
          "districtsBeingCompared": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "1",
        "nodeType": "class",
        "props": Object {
          "academy 20": Object {},
          "compareDistrict": undefined,
          "district": Object {
            "academy 20": Object {},
          },
          "districtsBeingCompared": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "2",
        "nodeType": "class",
        "props": Object {
          "agate 300": Object {},
          "compareDistrict": undefined,
          "district": Object {
            "agate 300": Object {},
          },
          "districtsBeingCompared": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <DistrictCard
            colorado={Object {}}
            district={
              Object {
                "colorado": Object {},
              }
            }
          />,
          <DistrictCard
            academy 20={Object {}}
            district={
              Object {
                "academy 20": Object {},
              }
            }
          />,
          <DistrictCard
            agate 300={Object {}}
            district={
              Object {
                "agate 300": Object {},
              }
            }
          />,
        ],
        "className": "DistrictsContainer",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "0",
          "nodeType": "class",
          "props": Object {
            "colorado": Object {},
            "compareDistrict": undefined,
            "district": Object {
              "colorado": Object {},
            },
            "districtsBeingCompared": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "1",
          "nodeType": "class",
          "props": Object {
            "academy 20": Object {},
            "compareDistrict": undefined,
            "district": Object {
              "academy 20": Object {},
            },
            "districtsBeingCompared": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "2",
          "nodeType": "class",
          "props": Object {
            "agate 300": Object {},
            "compareDistrict": undefined,
            "district": Object {
              "agate 300": Object {},
            },
            "districtsBeingCompared": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
